cdap-ui/app/cdap/components/AbstractWidget/SqlSelectorWidget/SchemaContainer.tsx

```diff
@@ -0,0 +1,220 @@
+      marginLeft: '5px',
+    },
+    tableRowTopBorder: {
+      borderTop: `1px solid ${theme.palette.grey[400]}`,
+    },
+    badgeDanger: {
+      backgroundColor: theme.palette.red[200],
+      color: 'white',
+      fontSize: '9px',
+      marginLeft: '5px',
+    },
+    errorRow: {
+      '& td': {
+        border: '0px',
+      },
+      backgroundColor: '#fde2e2',
+      border: `1px solid ${theme.palette.red[200]}`,
+    },
+  };
+};
+
+interface ISchemaContainerProps extends WithStyles<typeof styles> {
+  stage: IParsedInputSchema;
+  onExpandClick: (arg0: any) => void;
+  onSchemaChange: (arg0: any) => void;
+  aliases: object;
+  errorCount: number;
+}
+
+interface ISchemaContainerState {
+  selectFields: string;
+  menuAnchor: any;
+}
+
+class SchemaContainer extends React.Component<ISchemaContainerProps, ISchemaContainerState> {
+  public state = {
+    selectFields: 'All',
+    menuAnchor: null,
+  };
+
+  public tableSelectChange = (selectFields) => {
+    this.setState({ selectFields }, () => {
+      const newSchema = this.props.stage.schema.map((field) => {
+        return {
+          ...field,
+          selected: this.state.selectFields === 'All' ? true : false,
+        };
+      });
+      this.props.onSchemaChange(this.constructNewStage(newSchema));
+    });
+  };
+
+  public handleMenuClose = (value) => {
+    this.setState({ menuAnchor: null }, () => {
+      if (value) {
+        this.tableSelectChange(value);
+      }
+    });
+  };
+
+  public handleMenuOpen = (event) => {
+    this.setState({ menuAnchor: event.currentTarget });
+  };
+
+  public constructNewStage = (newSchema) => {
+    return {
+      ...this.props.stage,
+      schema: newSchema,
+    };
+  };
+
+  public toggleExpansionPanel = () => {
+    this.props.onExpandClick(this.props.stage);
+  };
+
+  public updateField = (newField) => {
+    if (this.props.stage && this.props.stage.schema) {
+      const newSchema = this.props.stage.schema.map((field) => {
+        if (field.name === newField.name) {
+          return newField;
+        }
+        return field;
+      });
+      this.props.onSchemaChange(this.constructNewStage(newSchema));
+    }
+  };
+
+  public render() {
+    const { classes } = this.props;
+    return (
+      <ExpansionPanel
+        className={classes.schemaContainer}
+        expanded={this.props.stage.expanded}
+        onChange={this.toggleExpansionPanel}
+      >
+        <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
+          <Typography variant="h6">{this.props.stage.name}</Typography>
+          <If condition={Boolean(this.props.errorCount)}>
+            <span className={classnames('badge', classes.badgeDanger)}>
+              {this.props.errorCount}
+            </span>
+          </If>
+        </ExpansionPanelSummary>
+        <ExpansionPanelDetails>
+          <Table>
+            <TableHead>
+              <TableRow>
+                <TableCell className={classes.schemaTableHeader} align="center">
```

Comment:
I think this should be `align="left"` Right now the heading & the contents don't align. Might be worth checking with Lea if that is ok.